homeassistant/components/enphase_envoy/sensor.py

```diff
@@ -414,11 +488,46 @@ def __init__(
         self._attr_device_info = DeviceInfo(
             identifiers={(DOMAIN, self.envoy_serial_num)},
             manufacturer="Enphase",
-            model=coordinator.envoy.part_number or "Envoy",
+            model=self.envoy_model(),
             name=coordinator.name,
             sw_version=str(coordinator.envoy.firmware),
+            hw_version=coordinator.envoy.part_number,
+            serial_number=self.envoy_serial_num,
         )
 
+    def envoy_model(self) -> str:
```

Comment:
Can we move this into the library?  It seems like HA shouldn't need to know how to generate this

Reply:
Yes. What's best path forward, leave as is for now until we added it to the library and replace when library is bumped. Or just name it 'Envoy' as it was and later apply the change when library is updated.

Reply:
It would be better to update the library, bump it in HA in another PR, and than rebase this PR

Reply:
Ok, I'll move it to the library first. My rebase track record is awfull with last 3 attempts spoiled so I'm not sure you should trust me with that action.

Reply:
No need to actually re-base, it's usually better just to merge commit in, or hit the update branch button. re-base and fourth push pattern is not desirable because it makes reviews harder, and since we're going to squash anyway there is no benefit to doing it

Reply:
Removed it and for now made fail safe with hasattr test and if not use 'Envoy'